### PR TITLE
New Adapter: Adsinteractive

### PR DESCRIPTION
**Pull request by balintvargha**
Dear Prebid Team!

This is our first attempt on registering an adapter to the Prebid Server. Please review the code, and send us any suggestions you seem appropriate. Thank you in advance

### CODE REVIEW

File: adapters/adsinteractive/adsinteractive.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	endpoint string
+}
+
+// Builder builds a new instance of the Foo adapter for the given bidder with the given config.
```
Comment (SyntaxNode) [inline adapters/adsinteractive/adsinteractive.go:19]: Thank you for following our adapter dev guide. Please replace Foo here with Adsinteractive, or your company casing AdsInteractive?

File: adapters/adsinteractive/adsinteractive_test.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+	"github.com/prebid/prebid-server/adapters/adapterstest"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+const testsDir = "adsinteractivetest"
+const testsBidderEndpoint = "http://bid.adsinteractive.com/prebid"
+
+func TestJsonSamples(t *testing.T) {
+	bidder, buildErr := Builder(openrtb_ext.BidderAdtarget, config.Adapter{
```
Comment (SyntaxNode) [inline adapters/adsinteractive/adsinteractive_test.go:15]: Please change the bidder name from BidderAdtarget to your new bidder name.

File: static/bidder-info/adsinteractive.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+endpoint: "http://bid.adsinteractive.com/prebid"
+maintainer:
+  email: it@adsinteractive.com
+gvlVendorID: 42
```
Comment (SyntaxNode) [inline static/bidder-info/adsinteractive.yaml:4]: Please either change this to your IAB GVL id, or remove if your company is not registered.

File: adapters/adsinteractive/adsinteractivetest/exemplary/banner.json [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+      }
+    }
+  ],
+
+  "expectedBidResponses": [
+    {
+      "currency": "EUR",
+      "bids": [
+        {
+          "bid": {
```
Comment (mansinahar) [inline adapters/adsinteractive/adsinteractivetest/exemplary/banner.json:73]: In general the PR looks to be in a good shape to me. The only thing I'd say is that please add JSON tests for the failure cases as well in the `adsinteractivetest/supplemental` directory. Like for 204, 400 and any other non-200 response.

File: adapters/adsinteractive/params_test.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+	}
+
+	for _, validParam := range validParams {
+		if err := validator.Validate(openrtb_ext.BidderAdsinteractive, json.RawMessage(validParam)); err != nil {
+			t.Errorf("Schema rejected adsinteractive params: %s", validParam)
+		}
+	}
+}
+
+// TestInvalidParams makes sure that the adtelligent schema rejects all the imp.ext fields we don't support.
```
Comment (onkarvhanumante) [inline adapters/adsinteractive/params_test.go:25]: Please  rename bidder name  from `adtelligent` to  `adsinteractive`

File: adapters/adsinteractive/adsinteractive.go [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+		}
+		return nil, []error{err}
+	}
+
+	if responseData.StatusCode != http.StatusOK {
+		err := &errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info.", responseData.StatusCode),
+		}
+		return nil, []error{err}
+	}
```
Comment (onkarvhanumante) [inline adapters/adsinteractive/adsinteractive.go:59]: should add JSON tests for the failure cases in the `adsinteractivetest/supplemental` directory

**Review by mansinahar — APPROVED** [review]

File: adapters/adsinteractive/params_test.go [hunk 7]
```diff
@@ -22,7 +22,7 @@ func TestValidParams(t *testing.T) {
 	}
 }
 
-// TestInvalidParams makes sure that the adtelligent schema rejects all the imp.ext fields we don't support.
+// TestInvalidParams makes sure that the Adsinteractive schema rejects all the imp.ext fields we don't support.
```
Comment (onkarvhanumante) [inline adapters/adsinteractive/params_test.go:25]: 👍🏼

**Review by onkarvhanumante — APPROVED** [review]

**Comment by balintvargha** [comment]
Dear Prebid Team!
Great news! Thank you for your help in the process

**Pull request merged**